lib/galaxy/tools/parser/output_objects.py

```diff
@@ -232,6 +277,26 @@ def collection_prototype(self, inputs, type_registry):
             collection_prototype.collection_type = collection_type
         return collection_prototype
 
+    def to_dict(self):
+        return {
+            'collection_type': self.collection_type,
+            'collection_type_source': self.collection_type_source,
+            'collection_type_from_rules': self.collection_type_from_rules,
+            'structured_like': self.structured_like,
+            'discover_datasets': list(map(lambda d: d.to_dict(), self.dataset_collector_descriptions)),
```

Comment:
```suggestion
            'discover_datasets': [d.to_dict() for d in self.dataset_collector_descriptions],
```
is easier on my eyes but completely fine either way. Probably not worth re-running the tests for.